lib/trino-geospatial-toolkit/src/main/java/io/trino/geospatial/GeometryUtils.java

```diff
@@ -160,4 +177,56 @@ public static boolean isPointOrRectangle(OGCGeometry ogcGeometry, Envelope envel
 
         return true;
     }
+
+    public static org.locationtech.jts.geom.Geometry jtsGeometryFromJson(String json)
+    {
+        verifyNonEmptyCoordinatesForAtomicTypes(json);
+        try {
+            return new GeoJsonReader().read(json);
+        }
+        catch (ParseException | IllegalArgumentException e) {
+            throw new TrinoException(INVALID_FUNCTION_ARGUMENT, "Invalid GeoJSON: " + e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Throws if json has empty coordinates for atomic type (Point, LineString, and Polygon).
+     * This is due to the inconsistent behavior in the locationtech library not returning
+     * an EMPTY type, but instead doing the following:
+     * {type:Point, coordinates:[]} -> POINT (0 0)
+     * {type:LineString, coordinates[]} -> LINESTRING EMPTY
+     * {type:Polygon, coordinates[]} -> Exception during parsing
+     *
+     * @param json
+     * @throws TrinoException
+     */
+    private static void verifyNonEmptyCoordinatesForAtomicTypes(String json)
+    {
+        try {
+            JsonNode jsonNode = OBJECT_MAPPER.readTree(json);
+            JsonNode typeNode = jsonNode.get(TYPE_ATTRIB);
+            if (typeNode != null) {
+                String type = typeNode.textValue();
+                if (ATOMIC_GEOMETRY_TYPES.contains(type)) {
+                    JsonNode coordinatesNode = jsonNode.get(COORDINATES_ATTRIB);
+                    if (coordinatesNode != null && coordinatesNode.isArray() && coordinatesNode.isEmpty()) {
+                        throw new TrinoException(INVALID_FUNCTION_ARGUMENT, format("Invalid GeoJSON: Empty %s not allowed.", type));
+                    }
+                }
+            }
+        }
+        catch (JsonProcessingException e) {
+        }
+    }
+
+    public static Optional<String> jsonFromJtsGeometry(org.locationtech.jts.geom.Geometry geometry)
+    {
+        // Throw if empty atomic type instead of returning invalid JSON in the form:
+        // {type:Point, coordinates:, ...}
+        if (ATOMIC_GEOMETRY_TYPES.contains(geometry.getGeometryType()) && geometry.isEmpty()) {
+            throw new TrinoException(INVALID_FUNCTION_ARGUMENT, format("Invalid GeoJSON: Empty %s not allowed.", geometry.getGeometryType()));
```

Comment:
nit: since GeoJSON is the output here, may be change `Invalid GeoJSON` to `Unsupported input` in error message?